spec/models/miq_product_feature_spec.rb

```diff
@@ -16,19 +14,48 @@
     )
   end
 
+  def traverse_product_features(product_feature, &block)
+    block.call(product_feature)
+    if product_feature.key?(:children)
+      product_feature[:children].each { |child| traverse_product_features(child, &block) }
+    end
+  end
+
+  RSpec::Matchers.define :have_product_feature_structure do
+    match do |actual|
+      return false if (actual.keys - possible_keys).any?
+      return false if (required_keys - actual.keys).any?
+      return false if actual.key?(:children) && actual[:children].empty?
+      true
+    end
+
+    diffable
+
+    def required_keys
+      [:identifier]
+    end
+
+    def possible_keys
+      required_keys + [:name, :feature_type, :description, :children, :hidden, :protected]
+    end
+  end
+
+  it "is properly configured" do
+    everything = YAML.load_file(described_class.feature_yaml)
+    traverse_product_features(everything) { |pf| expect(pf).to have_product_feature_structure }
+  end
+
   context ".seed" do
     it "creates feature identifiers once on first seed, changes nothing on second seed" do
-      status_seed1 = MiqProductFeature.seed
-      expect(MiqProductFeature.count).to eq(expected_feature_count)
-      expect(status_seed1[:created]).to match_array status_seed1[:created].uniq
-      expect(status_seed1[:updated]).to match_array []
-      expect(status_seed1[:unchanged]).to match_array []
-
-      status_seed2 = MiqProductFeature.seed
-      expect(MiqProductFeature.count).to eq(expected_feature_count)
-      expect(status_seed2[:created]).to match_array []
-      expect(status_seed2[:updated]).to match_array []
-      expect(status_seed2[:unchanged]).to match_array status_seed1[:created]
+      expect { @status_seed1 = MiqProductFeature.seed }.to change(MiqProductFeature, :count)
+      expect(@status_seed1[:created]).to match_array @status_seed1[:created].uniq
+      expect(@status_seed1[:updated]).to match_array []
```

Comment:
Note, typically this would fail if you accidentally use a same identifier on two or more features since we find by identifier and update the record.  

first seed...
feature_1_with_identifier_foo   # creates the record
feature_2_with_identifier_foo  # updates the record